crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -12,31 +11,30 @@ pub struct CastConfig {
     pub rpc_url: String,
     pub network: String,
     pub account: String,
+    pub accounts_file: Option<Utf8PathBuf>,
 }
 
-pub fn get_property(
-    tool: &Option<BTreeMap<String, Value>>,
+pub fn get_property(tool: &Value, property: &str) -> String {
+    tool.get(property)
+        .and_then(serde_json::Value::as_str)
+        .map(String::from)
+        .unwrap_or_default()
+}
+
+pub fn get_property_from_profile(
+    tool_sncast: &Value,
```

Comment:
what is the reason to have two functions?

Reply:
My reasoning was that both are smaller, hence they are easier to comprehend. The abstraction split works great here imo

Reply:
maybe, but it seems to me that in `get_tools` you only return some field from tool, and the function is used in one place? so it perhaps could make sense to just do it in one place?

Reply:
If we talk about `get_tools`, it's used in 2 places (parse_scarb_config + add_created_profile_to_configuration). Also, seeing a call to function named `get_package_tool_sncast` immediately tells me that it's getting the reference to the [tool.sncast] and I don't need to know how it gets it and what are the errors it could return, making the code cleaner in my opinion.

Besides, breaking things down into smaller functions (even if they are used once) does not in fact increase the amount of code (besides the lines with the funcition's signature 😸) and makes the outer functions easier to grasp